src/main/java/com/mucommander/ui/autocomplete/TextFieldCompletion.java

```diff
@@ -77,6 +77,11 @@ public TextFieldCompletion(AutocompleterTextComponent comp, Completer completer)
 
     	autocompletedtextComp.getDocument().addDocumentListener(documentListener);
 
+        // In previous versions of muCommander, the 'home' and 'end' keys were forwarded
+        // to the pop-up list (if opened), which I found surprising. This version handles
+        // the 'home' and 'end' keys always in the text field, like Firefox and others do.
+        boolean SHOULD_HOME_AND_END_KEYS_WORK_ON_POPUP_LIST = false;
```

Comment:
About your first suggestion, what you propose is not intuitive. You may argue that this is completely subjective, but it is not the way Firefox, Chrome or Dolphin behave.

About your second point, if we do decide later to make it configurable, the flag is already there, so why throw away working code? You can remove it, if you wish, but please do not make me rework the pull request for such a trivial amendment, as it would take much longer than to modify the code in your editor and commit it!